apps/proxy/src/app/create-project/create-project.component.ts

```diff
@@ -0,0 +1,208 @@
+    public environments_with_slug;
+    public environmentParams = {
+        itemsPerPage: 5,
+        pageNo: 1,
+    };
+    public projects$: Observable<IPaginatedResponse<IProjects[]>>;
+    public environments$: Observable<IPaginatedResponse<IEnvironments[]>> = this.componentStore.environments$;
+    public sourceDomain$: Observable<any> = this.componentStore.sourceDomain$;
+    public getProject$: Observable<boolean> = this.componentStore.getProject$;
+    public isLoading$: Observable<boolean> = this.componentStore.isLoading$;
+    public projectId: number;
+
+    constructor(
+        private componentStore: CreateProjectComponentStore,
+        private fb: FormBuilder,
+        private store: Store<IAppState>
+    ) {
+        super();
+        this.projects$ = this.store.pipe(select(selectAllProjectList));
+        this.primaryDetailsForm = this.fb.group({
+            name: [
+                '',
+                [
+                    Validators.required,
+                    Validators.pattern(CAMPAIGN_NAME_REGEX),
+                    CustomValidators.minLengthThreeWithoutSpace,
+                    CustomValidators.noStartEndSpaces,
+                    Validators.maxLength(20),
+                ],
+            ],
+            rateLimitHit: [null, [Validators.required, Validators.pattern(ONLY_INTEGER_REGEX)]],
+            rateLimitMinute: [null, [Validators.required, Validators.pattern(ONLY_INTEGER_REGEX)]],
+            selectedEnvironments: [''],
+        });
+
+        (this.destinationUrlForm = this.fb.group({
+            endpoint: ['', Validators.required],
+            ForwardUrl: this.fb.array([]),
+        })),
+            (this.gatewayUrlDetailsForm = this.fb.group({
+                useSameUrlForAll: [false],
+                gatewayUrls: this.fb.array([]),
+            }));
+    }
+
+    ngOnInit(): void {
+        this.getEnvironment();
+        this.projects$.pipe(takeUntil(this.destroy$)).subscribe((res) => {
+            if (res) {
+                const latestProject = res.data.slice().reverse();
+
+                latestProject.forEach((project) => {
+                    const baseUrl = `${environment.baseUrl}/proxy`;
+                    this.projectId = project.id;
+                    this.environments_with_slug = project.environments_with_slug.map((res) => ({
+                        name: res.name,
+                        url: `${baseUrl}/${this.projectId}/${res.project_slug}`,
+                    }));
+                });
+                this.populateGatewayUrls();
+                this.populateForwardUrls();
+            }
+        });
+        this.getProject$.pipe(takeUntil(this.destroy$)).subscribe((res) => {
+            if (res) {
+                this.currentstep = 2;
+                this.store.dispatch(rootActions.getAllProject());
+            }
+        });
+    }
+
+    get gatewayUrls(): FormArray {
+        return this.gatewayUrlDetailsForm.get('gatewayUrls') as FormArray;
```

Comment:
When you Add Interface in form create Interface for FormArray separately and use it here and also in main Interface 
for both gatewayUrls and forwardUrls